presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -202,7 +206,11 @@ public NodeSelector createNodeSelector(ConnectorId connectorId, int maxTasksPerS
                 }
             }
 
-            return new NodeMap(byNodeId.build(), byHostAndPort.build(), byHost.build(), workersByNetworkPath.build(), coordinatorNodeIds);
+            for (InternalNode node : inactiveNodes) {
```

Comment:
1. Yes the orchestration service brings up a new node then this solution would make cache ineffective, until it resets. Usually happens within 10 mins or so in a busy system.
2. We have noticed that repeat offender, kernel upgrades, does bring the same node back. Which is the problem space for this PR. 
3. Consistent hashing is indeed a good solution for the cases where capacity is removed or added. For this cases, the capacity gets reset within 10 mins or so.
4. I am not sure on the configuration of total number of nodes. It would get messy with the nodes addition or removal.

Reply:
Right. Configuring the total number of nodes would hurt the flexibility a lot.

How about for this PR, when a node disappears, we mark it as "temporarily unavailable" and it does not shift the total number of nodes so the cache shifting does not happen, and only remove it from the pool when a certain amount of time has passed. I think it would help cover "same node" issue and it is the scope for this PR.

The downside of course is that if indeed a new node is back, we might have to shift the cache twice (when a new node is added, and when the old node is eventually removed), but I guess it is fine because today we also shift it twice anyway :)

Reply:
Somehow I missed this. Yeah, thats the idea.